src/modm/platform/adc/stm32f3/adc_impl.hpp.in

```diff
@@ -142,31 +143,96 @@ modm::platform::Adc{{ id }}::setLeftAdjustResult(const bool enable)
 }
 
 bool
-modm::platform::Adc{{ id }}::setChannel(	const Channel channel,
-										const SampleTime sampleTime)
+modm::platform::Adc{{ id }}::setChannel(const nonstd::span<const Channel> channels)
+{
+	if(channels.size() >= maxSequenceLength) {
+		return false;
+	}
+
+	// clear SQRx registers
+	ADC{{ id }}->SQR1 = 0;
+	ADC{{ id }}->SQR2 = 0;
+	ADC{{ id }}->SQR3 = 0;
+	ADC{{ id }}->SQR4 = 0;
+
+	for(unsigned int i = 0; i < channels.size(); i++) {
+		if(!setSequenceChannel(i, channels[i])) {
+			return false;
+		}
+	}
+	ADC{{ id }}->SQR1 |= channels.size();
+	return true;
+}
+
+bool
+modm::platform::Adc{{ id }}::setInjectedChannel(const nonstd::span<const Channel> channels)
+{
+	if(channels.size() >= maxInjectedSequenceLength) {
+		return false;
+	}
+	// clear JSQR register except for JEXTEN and JEXTSEL bits
+	ADC{{ id }}->JSQR &= ADC_JSQR_JEXTEN_Msk | ADC_JSQR_JEXTSEL_Msk;
```

Comment:
```suggestion
	ADC{{ id }}->JSQR &= ~(ADC_JSQR_JEXTEN_Msk | ADC_JSQR_JEXTSEL_Msk);
```

Reply:
Or 
```suggestion
	ADC{{ id }}->JSQR = ADC_JSQR_JEXTEN_Msk | ADC_JSQR_JEXTSEL_Msk;
```

Reply:
Nope, as the comment above states: *clear JSQR register except for JEXTEN and JEXTSEL bits*.